### PR TITLE
Add support for the mRo Control Zero F7, a new flight controller from mRobotics

### PR DESCRIPTION
**Pull request by pkocmoud**
Bench tested on hardware, ready for flight testing.

### CODE REVIEW

File: libraries/AP_Baro/AP_Baro_DPS280.cpp [hunk 1]
```diff
@@ -126,6 +126,7 @@ bool AP_Baro_DPS280::init()
         return false;
     }
 
+    dev->set_read_flag(0x80);
```
Comment (sh83) [inline libraries/AP_Baro/AP_Baro_DPS280.cpp]: Should be used only if  sensor on spi bus.

File: libraries/AP_Compass/AP_Compass.cpp [hunk 2]
```diff
@@ -938,6 +938,8 @@ void Compass::_detect_backends(void)
     ADD_BACKEND(DRIVER_LSM9DS1, AP_Compass_LSM9DS1::probe(hal.spi->get_device("lsm9ds1_m")));
 #elif HAL_COMPASS_DEFAULT == HAL_COMPASS_BMM150_I2C
     ADD_BACKEND(DRIVER_BMM150, AP_Compass_BMM150::probe(GET_I2C_DEVICE(HAL_COMPASS_BMM150_I2C_BUS, HAL_COMPASS_BMM150_I2C_ADDR)));
+#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_CHIBIOS_MROCONTROLZEROF7
```
Comment (sh83) [inline libraries/AP_Compass/AP_Compass.cpp]: Why do you use board_subtype and not HAL_COMPASS_DEFAULT?

File: libraries/AP_InertialSensor/AP_InertialSensor_Invensensev2.cpp [hunk 3]
```diff
@@ -639,9 +640,8 @@ bool AP_InertialSensor_Invensensev2::_hardware_init(void)
         return false;
     }
 
-    // setup for register checking. We check much less often on I2C
-    // where the cost of the checks is higher
-    _dev->setup_checked_registers(7, _dev->bus_type() == AP_HAL::Device::BUS_TYPE_I2C?200:20);
+    // disabled setup of checked registers as it can't cope with bank switching
+    // _dev->setup_checked_registers(7, _dev->bus_type() == AP_HAL::Device::BUS_TYPE_I2C?200:20);
```
Comment (sh83) [inline libraries/AP_InertialSensor/AP_InertialSensor_Invensensev2.cpp]: The fix already  merged, didnt it?

File: libraries/AP_Compass/AP_Compass.cpp (hunk 2 above)
Comment (pkocmoud) [reply libraries/AP_Compass/AP_Compass.cpp]: @sh83 The sensor orientation is specific to this board so why not use CONFIG_HAL_BOARD_SUBTYPE

File: libraries/AP_InertialSensor/AP_InertialSensor_Invensensev2.cpp (hunk 3 above)
Comment (pkocmoud) [reply libraries/AP_InertialSensor/AP_InertialSensor_Invensensev2.cpp]: @sh83 this PR was based of yesterday's master, so do you have a PR to reference where it was previously merged?

File: libraries/AP_Baro/AP_Baro_DPS280.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_Baro/AP_Baro_DPS280.cpp]: its harmless on i2c. We normally put it in unconditionally, so it works for both bus types

File: libraries/AP_Compass/AP_Compass.cpp (hunk 2 above)
Comment (tridge) [reply libraries/AP_Compass/AP_Compass.cpp]: also, HAL_COMPASS_DEFAULT doesn't allow for multiple sensors

File: libraries/AP_InertialSensor/AP_InertialSensor_Invensensev2.cpp (hunk 3 above)
Comment (tridge) [reply libraries/AP_InertialSensor/AP_InertialSensor_Invensensev2.cpp]: it wasn't merged yet, but it is now

File: libraries/AP_Compass/AP_Compass.cpp (hunk 2 above)
Comment (sh83) [reply libraries/AP_Compass/AP_Compass.cpp]: Orientation is also could set by define. See HAL_COMPASS_HMC5843_ROTATION

**Pull request merged**